### DISCUSSION TITLE
20241030 Weekly Development Meeting

### DISCUSSION THREAD

**Discussion by josecelano** [nautilus-cyberneering/.github#114] [Meetings]
- Time:  9:30 in Canary Islands (WET (UTC+0)).
- Time: 10:30 in Vaduz (CET (UTC+1)).

Location:
<https://meeting.blockfinance-eco.li/NautilusCyberneeringWeeklyDevMeeting>

Expected duration: 2 to 4 hours with breaks

Communicated attendants:

- Cameron Garnham(@da2ce7)  - x 
- Jose Celano(@josecelano) - x
- Constantin Bosse(@cgbosse) - x
- Mario Melchor(@mario-nt) - x
- Graeme Byrne(@grmbyrn) - x

-------------------------------------------------------------

## Agenda:

1. Everyone will take their turn and:
- Go over his notes added as comments for the meeting which will have been added in advance to give an update of all the work done during the previous week indicating.
	- The issues worked on
	- The progress made on them
	- Other things done
- Inform of any faced challenges and request help if necessary to work on specific issues.
- Highlight the encountered difficulties, especially if there is a risk of them causing delays or affecting end product's quality.
- Communicate what will be the next week's tasks.
- List any new topics for further discussion in an additional Technical meeting that he noted for future discussion in comments to this discussion either before the meeting or after his turn so that they get considered.

During the meeting every team member will take their turns and follow this sequence.
The topics for further discussion will be noted in advance by each the team member and to the meetings agenda.

2. At the end of the meeting if there any new technical topics to be discussed further it will be decided to
set another meeting date for the discussion of these or to hold the meeting hereafter.

Also the persons to be part of the meeting will be decided on so that the others will be free to continue their own work.

**Comment by josecelano** [comment]
### Events (14.40h)

Rust Las Palmas meetup: https://rust-laspalmas.dev/es/enyesque-2024-10-review/

### Testing in Rust (1.75h)

New article: [Testing error variants in Rust using matches! macro](https://github.com/nautilus-cyberneering/testing-in-rust/pull/5)

### Torrust (11.5h)

#### Torrust Website 

New articles:

- [Contributor path](https://torrust.com/blog/contributor-path)
- [Bencode to JSON Converter in Rust](https://github.com/torrust/torrust-website/pull/103) (PR pending to merge, waiting for publishing the crate).

PR review:

- https://github.com/torrust/torrust-website/pull/102

#### Torrust Tracker

Issues:

- [ ] https://github.com/torrust/torrust-tracker/issues/360 

#### Torrust Index

Issues:

- [ ] https://github.com/torrust/torrust-index/issues/251

#### Bittorrent Primitives

I created this new package to extract the duplicate InfoHash type from the Tracker and Index.

https://github.com/torrust/bittorrent-primitives

### Other Projects (7.5h)

**Comment by grmbyrn** [comment]
Torrust Website:
* [Add a path segment in the URL for sections #73](https://github.com/torrust/torrust-website/issues/73)
* [Ask the designer about the blog post card #60](https://github.com/torrust/torrust-website/issues/60)
* [Top Left Torrust Logo-Home-Button Visual Feedback #93](https://github.com/torrust/torrust-website/issues/93)
* [Bottom Nautilus Logo is not a link, and has no visual feedback it it should be link. #95](https://github.com/torrust/torrust-website/issues/95)
* [Latest articles in homepage are not ordered #105](https://github.com/torrust/torrust-website/issues/105)
* Fix links to articles in Related Posts under each blog post.

Learn About Migrating to Svelte 5:
* [Svelte 5 is alive](https://svelte.dev/blog/svelte-5-is-alive)
* [Svelte 5 migration guide](https://svelte.dev/docs/svelte/v5-migration-guide)
* [The Svelte 5 Migration Guide](https://www.youtube.com/watch?v=_q5FZ3Z9Fz8&ab_channel=JoyofCode)

**Comment by mario-nt** [comment]
### GUI
- https://github.com/torrust/torrust-index-gui/pull/629

### Index

- https://github.com/torrust/torrust-index/pull/754

### Review

- https://github.com/torrust/torrust-index/pull/755#pullrequestreview-2397147001
- https://github.com/torrust/bencode2json
